packages/contracts-bedrock/src/L1/OPPrestateUpdater.sol

```diff
@@ -0,0 +1,168 @@
+ */
+contract OPPrestateUpdater is OPContractsManager {
+    struct PrestateUpdateInput {
+        OpChain opChain;
+        Claim absolutePrestate;
+        bool newFDG;
+        address pdgProposer;
+        address pdgChallenger;
+        address vm;
+        uint256 initBond;
+    }
+
+    /**
+     * @notice Returns the version string for this contract
+     * @return Version string
+     */
+    function version() public pure override returns (string memory) {
+        return "1.0.0";
+    }
+
+    /**
+     * @notice Constructs the CustomOPContractsManager contract
+     * @param _superchainConfig Address of the SuperchainConfig contract
+     * @param _protocolVersions Address of the ProtocolVersions contract
+     * @param _l1ContractsRelease Version string for L1 contracts release
+     * @param _blueprints Addresses of Blueprint contracts
+     * @param _implementations Addresses of implementation contracts
+     * @param _upgradeController Address of the upgrade controller
+     */
+    constructor(
+        ISuperchainConfig _superchainConfig,
+        IProtocolVersions _protocolVersions,
+        string memory _l1ContractsRelease,
+        Blueprints memory _blueprints,
+        Implementations memory _implementations,
+        address _upgradeController
+    )
+        OPContractsManager(
+            _superchainConfig,
+            _protocolVersions,
+            _l1ContractsRelease,
+            _blueprints,
+            _implementations,
+            _upgradeController
+        )
+    { }
+
+    /// @notice Overrides the l1ContractsRelease function to return "none", as this OPCM
+    /// is not releasing new contracts.
+    function l1ContractsRelease() external pure override returns (string memory) {
+        return "none";
+    }
+
+    function deploy(DeployInput memory) external pure override returns (DeployOutput memory) {
+        revert("Not implemented");
+    }
+
+    function upgrade(OpChain[] memory) external pure override {
+        revert("Not implemented");
+    }
+
+    function addGameType(AddGameInput[] memory) public pure override returns (AddGameOutput[] memory) {
+        revert("Not implemented");
+    }
+
+    /// @notice Updates the prestate hash for a new game type while keeping all other parameters the same
+    /// @param _prestateUpdateInputs The new prestate hash to use
+    function updatePrestate(PrestateUpdateInput[] memory _prestateUpdateInputs) external {
+        // Loop through each chain and prestate hash
+        for (uint256 i = 0; i < _prestateUpdateInputs.length; i++) {
+            bool hasFDG = Claim.unwrap(_prestateUpdateInputs[i].absolutePrestate) != bytes32(0);
+            AddGameInput[] memory inputs = new AddGameInput[](hasFDG ? 2 : 1);
+            AddGameInput memory pdgInput;
+            AddGameInput memory fdgInput;
+
+            if (Claim.unwrap(_prestateUpdateInputs[i].absolutePrestate) == bytes32(0)) {
```

Comment:
How can we use `Claim.unwrap(_prestateUpdateInputs[i].absolutePrestate) != bytes32(0)` to determine if there is a FDG when we also revert if `Claim.unwrap(_prestateUpdateInputs[i].absolutePrestate) == bytes32(0)`?

I'd have expected we determine if there's a FDG by seeing if `DisputeGameFactory.gameImpls(GameTypes.CANNON) != address(0)`.

Reply:
Agree, that check looks wrong. Should be similar to [here](https://github.com/ethereum-optimism/optimism/blob/develop/packages/contracts-bedrock/src/L1/OPContractsManager.sol#L590-L594)

Reply:
OK, I see. This line dates to when we were accepting an [absolutePrestate for each game](https://github.com/ethereum-optimism/optimism/pull/13998/files/bd5e9a9876e66909b87edd18156eb80d3c6cedcb..3565fbb55a7a3cd8a3d40747d58b3121457aab0a#diff-723054ee19c5dac147d1e3480872f2826eb8b50982d0feaf0ec7db981fdf010dL88), and so providing one for the fdg was just taken as an indication that it exists. 

Accepting a single `absolutePrestate`, then querying to see if there is an FDG is a better approach.

Reply:
This has been addressed: 

https://github.com/ethereum-optimism/optimism/blob/ad9d9b4546115f8d74b19f859499df5ea10389cc/packages/contracts-bedrock/src/L1/OPPrestateUpdater.sol#L99